RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -464,8 +464,17 @@ class PumpOpsSynchronous {
                     throw error
                 }
             }
-            
-            NSLog("Fetched page %d: %@", pageNum, pageData as NSData)
+            if true {
+                let logString = NSString(format: "Fetched page %d: %@", pageNum, pageData as NSData ) 
+                let sz = logString.length
```

Comment:
Iterating over a string is thorny due to multibyte character support. Iterating over `Data` is simple. Could you change this loop to iterate over blocks of `pageData` instead?

Also, the length of `pageData` is always 2,304 characters, which you could rely on.